packages/toolpad-app/src/toolpad/propertyControls/SelectOptions.tsx

```diff
@@ -81,11 +103,24 @@ function SelectOptionsPropEditor({
 
   const handleOptionChange = React.useCallback(
     (newOption: string | SelectOption) => {
+      const errorState = value.some((item) =>
```

Comment:
We should be able to reuse the same function here as in `watchInputValue` above, can you please make a reusable function that is used in both places?

Also, repeated logic inside this function such as `(newOption as SelectOption)?.value ?? newOption` could be assigned to a variable ahead of time so you don't have to write it twice.